onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,121 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+/// @title Quorum Factory Test
+pragma solidity ^0.8.8;
+
+import {Test} from "forge-std/Test.sol";
+import {QuorumFactory} from "contracts/consensus/quorum/QuorumFactory.sol";
+import {Quorum} from "contracts/consensus/quorum/Quorum.sol";
+import {IHistory} from "contracts/history/IHistory.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+contract QuorumFactoryTest is Test {
+    QuorumFactory factory;
+    address[] validators;
+    uint256[] shares;
```

Comment:
These state variables can be removed. Instead of storage, they can be (and kind of already are, really) handled in memory! For example, `generateAddrAndShares` should return these, instead of writing them to storage. This should allow this function to be `pure` then.

Reply:
We could do that, although gas optimization is not an issue in test contract.

Reply:
It's not much about gas, but about where data is stored.
I think the code would look much cleaner if it were handled in memory.